lib/bolt/module_installer/puppetfile.rb

```diff
@@ -97,20 +97,43 @@ def assert_satisfies(specs)
           end
         end
 
-        return if unsatisfied_specs.empty?
-
+        #versionless_mods = @modules.any? do |mod|    
+        #  mod.version.nil? 
+        #end
+        versionless_mods = []
+        @modules.each {|mod|
+          if mod.version.nil?
+            versionless_mods << mod
+          end
+        }
```

Comment:
```suggestion
        versionless_mods = @modules.select { |mod| mod.version.nil? }
```

[Array#select](https://ruby-doc.org/core-2.7.1/Array.html#method-i-select) lets you create an array of items that meet a given condition.

Also, if the block is a single line then you can use the bracket notation. If the block is multiple lines, use `do...end`.

Reply:
Oops, actually, this needs to make sure the module is a Forge module, too. Git modules don't have a `version` method, so you'll get an "undefined method" error otherwise. Updated the suggestion above.